core/src/main/scala/org/apache/spark/status/LiveEntity.scala

```diff
@@ -254,6 +255,7 @@ private class LiveExecutor(val executorId: String, _addTime: Long) extends LiveE
   var totalShuffleRead = 0L
   var totalShuffleWrite = 0L
   var isBlacklisted = false
+  var blacklistedInStages: Set[Int] = TreeSet()
```

Comment:
any particular reason you chose `TreeSet`, and not just `scala.collection.immutable.Set` (scala has a default implementation, specialized for a small number of items, then goes to a `HashSet`) -- see use of `Map` as implementation for `executorLogs`.

Reply:
I have chosen it  because of the ordering as I assumed the user would like to see here a monoton not decreasing ordering of stageIDs. If it is not needed of course I can change it.

Reply:
yeah thats a good reason, sorry I should have thought of that!